util/formatter.cjs

```diff
@@ -0,0 +1,65 @@
+/**
+ * @prettier
+ */
+'use strict';
+const fs = require('fs');
+const path = require('path');
+
+const eslint = require('eslint');
+const prettier = require('prettier');
+
+const format = async (code, filepath) => {
+  const options = await prettier.resolveConfig(filepath);
+  return prettier.format(code, { filepath, ...options });
+};
+
+const linter = new eslint.ESLint({
```

Comment:
Is this file doing something different than running [prettier-eslint](https://github.com/prettier/prettier-eslint)?

Reply:
prettier-eslint didn't respect "@prettier" comment

Reply:
> I think in general we want to format everything, right? So any "prettier" comment is only temporary?

So you want to format anything prettier could format in one PR?

There is a `requirePragma` option that we could do it progressively. we can remove them when nearly all files is formatted with prettier.

Reply:
I was trying to ask if `requirePragma` is something you think we will need after everything has been formatted or if it is only a temporary option to make converting easier to do in smaller pieces.  It sounds like you're saying we can remove them when we're done.

I think I would rather do a separate formatting pass to finish formatting everything than have people formatting files as they are fixing bugs or converting to TypeScript.  It doesn't (shouldn't??) be one PR to do this formatting, but I would prefer that once this lands we format everything soon after.

Another option is to add directories to `.prettierignore` and remove them from there as you format progressively, rather than adding lines to each file.

Reply:
it's temporary.

so maybe do it module by module?

Reply:
Sorry for maybe missing a side conversation here, I don't think I understand what we're doing in this file vs running prettier-eslint?

Can this be used as a formatter in IDEs?

Reply:
@panicstevenson If you are using webstorm or IDEA, add a external tool.

If you are using vscode, I have no idea...